pkg/chunk/purger/blocks_purger_api.go

```diff
@@ -17,24 +17,30 @@ import (
 	cortex_tsdb "github.com/cortexproject/cortex/pkg/storage/tsdb"
 	"github.com/cortexproject/cortex/pkg/tenant"
 	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/validation"
 )
 
 type BlocksPurgerAPI struct {
 	bucketClient objstore.Bucket
 	logger       log.Logger
+	overrides    *validation.Overrides
 }
 
-func NewBlocksPurgerAPI(storageCfg cortex_tsdb.BlocksStorageConfig, logger log.Logger, reg prometheus.Registerer) (*BlocksPurgerAPI, error) {
+func NewBlocksPurgerAPI(storageCfg cortex_tsdb.BlocksStorageConfig, overrides *validation.Overrides, logger log.Logger, reg prometheus.Registerer) (*BlocksPurgerAPI, error) {
```

Comment:
I'd just use new `UserBucketOverrides` type, to better express dependencies and reduce unnecessary imports. (Applies to all modified files, where applicable)